packages/app/src/firebase.js

```diff
@@ -25,7 +25,8 @@ export function FirebaseAuthProvider({ children }) {
   useEffect(() => {
     firebase.auth().onAuthStateChanged(async user => {
       if (user) {
-        // User is signed in.
+        const { credential } = await firebase.auth().getRedirectResult()
```

Comment:
🎉🎉🎉
You can have the username without calling github api.

You just have to do:
```js
const { additionalUserInfo } = await firebase.auth().getRedirectResult()
console.log(additionalUserInfo.username)
```
You have also the property `additionalUserInfo.isNewUser` 🤪

And in `additionalUserInfo.profile` you have more info than the username: bio, company, number of followers, location, number of repos... 🤯

Reply:
Be aware that you will get these data only when the user click on the "Sign in with GitHub" button. When the session is kept in cookies and "automatically" connected, these data will be null. So you will need to store then if you need them.

Reply:
https://firebase.google.com/docs/reference/js/firebase.auth.html#additionaluserinfo